scripts/test-html.js

```diff
@@ -60,8 +61,40 @@ function waitComplete(dom) {
 }
 
 function serveDirectory(dir) {
-  const fileServer = new static.Server(dir);
-  const server = http.createServer((req, res) => fileServer.serve(req, res));
+  const types = {
+    '.html': 'text/html',
+    '.css': 'text/css',
+    '.js': 'application/javascript',
+  };
+
+  const root = path.normalize(path.resolve(dir));
+
+  const server = http.createServer((request, reply) => {
+    console.log(`${request.method} ${request.url}`);
+
+    const fileNotFound = () => {
+      reply.writeHead(404, { 'Content-Type': 'text/html' });
+      reply.end("404: File not found");
+    };
+
+    const ext = path.extname(request.url);
+    if (!types.hasOwnProperty(ext))
+      return fileNotFound();
+
+    const filePath = path.join(root, request.url);
+    const isPathUnderRoot = path.normalize(path.resolve(filePath)).startsWith(root);
+    if (!isPathUnderRoot)
+      return fileNotFound();
+
+    fs.readFile(filePath, (err, data) => {
+      if (err) {
+        fileNotFound();
+      } else {
+        reply.writeHead(200, { 'Content-Type': types[ext] });
+        reply.end(data);
+      }
+    })
+  });
```

Comment:
Have you considered:

```js
const app = express();
app.use(express.static(dir));
const server = http.createServer(app);
```

Reply:
Gosh, I was looking for something simple like that that was still well maintained. I think I overlooked Express because I thought it was a full-blown web framework.

I updated to use express as suggested instead.